presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/SphericalGeographyType.java

```diff
@@ -75,6 +76,9 @@ public Object getObjectValue(ConnectorSession session, Block block, int position
             return null;
         }
         Slice slice = block.getSlice(position, 0, block.getSliceLength(position));
+        if (slice == Slices.EMPTY_SLICE) {
+            return null;
+        }
```

Comment:
Remove this by returning null above.

Reply:
I don't think @SqlNullable is applicable here, the input is not null but rather the underlying geometry is empty. Also as mentioned above,  returning null would cause a crash in the SQL engine when trying to write the result (most likely due to the fact that the return type is not a primitive). I haven't been able to find an easy way to prevent the engine from calling SphericalGeographyType#writeSlice.

Reply:
From the `@SqlNullable` documentation (https://prestodb.io/docs/current/develop/functions.html):
> The method must be annotated with @SqlNullable if it can return NULL when the arguments are non-null.
Since you are having trouble returning null when the input is not null, and we return null when the input is not null in other functions decorated by `@SqlNullable`, I'd still try that.

Reply:
Read the doc, looks like I misinterpreted the semantic of SqlNullable. I confirm that adding SqlNullable enables me to return null, which is awesome.